aptos-move/framework/aptos-framework/sources/SharedAccount.move

```diff
@@ -0,0 +1,153 @@
+module AptosFramework::SharedAccount {
+    use Std::Errors;
+    use Std::Signer;
+    use Std::Vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+    use AptosFramework::SimpleMap::{Self, SimpleMap};
+    use AptosFramework::ResourceAccount;
+
+    struct Shares has store {
+        numerator: u64,
+        denominator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: SimpleMap<address, Shares>,
+    }
+
+    const EINVALID_INIT_INPUT: u64 = 0;
+    const EACCOUNT_DNE: u64 = 1;
+    const EADDRESS_ALREADY_EXISTS: u64 = 2;
+    const EINVALID_NUMERATOR_DENOMINATOR_COMBINATIONS: u64 = 3;
+    const EINVALID_SIGNER: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
+
+    public(script) fun initialize_shared_account(source: &signer, seed: vector<u8>): (signer) {
```

Comment:
this can probably just be a `fun` and not a script if it won't be called on it's own.

Also, probably not public since it can't stand on it's own without `initialize`

Reply:
ooh makes sense! I merged it with the `initialize()` function now